router/core/retry_expression_builder.go

```diff
@@ -0,0 +1,85 @@
+package core
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/wundergraph/cosmo/router/internal/expr"
+	"github.com/wundergraph/cosmo/router/internal/retrytransport"
+	"go.uber.org/zap"
+)
+
+const DefaultRetryExpression = "IsRetryableStatusCode() || IsConnectionError() || IsTimeout()"
+
+var noopRetryFunc = func(err error, req *http.Request, resp *http.Response) bool {
+	return false
+}
+
+// BuildRetryFunction creates a ShouldRetry function based on the provided expression
+func BuildRetryFunction(retryOpts retrytransport.RetryOptions) (retrytransport.ShouldRetryFunc, error) {
+	// We do not need to build a retry function if retries are disabled
+	// This means that any bad expressions are ignored if retries are disabled
+	if !retryOpts.Enabled {
+		return noopRetryFunc, nil
+	}
+
+	// Use default expression if empty string is passed
+	expression := retryOpts.Expression
+	if expression == "" {
+		expression = DefaultRetryExpression
+	}
+
+	// Create the retry expression manager
+	manager, err := expr.NewRetryExpressionManager(expression)
+	if err != nil {
+		return nil, fmt.Errorf("failed to compile retry expression: %w", err)
```

Comment:
NewRetryExpressionManager already has a `failed to compile` errors message. This will create a duplicate error message

Reply:
Done